specification/assets/gherkin/hooks.feature

```diff
@@ -0,0 +1,49 @@
+@hooks
+Feature: Evaluation details through hooks
+
+# This test suite contains scenarios to test the functionality of hooks.
+
+  Background:
+    Given a stable provider
+
+  Scenario: Passes evaluation details to after and finally hooks
+    Given a client with added hook
+    And a boolean-flag with key "boolean-flag" and a default value "false"
+    When the flag was evaluated with details
+    Then the "before" hook should have been executed
+    And the "after, finally" hooks should be called with evaluation details
+      | data_type | key        | value        |
+      | string    | flag_key   | boolean-flag |
+      | boolean   | value      | true         |
+      | string    | variant    | on           |
+      | string    | reason     | STATIC       |
+      | string    | error_code | None         |
+
+  # errors
+  Scenario: Flag not found
+    Given a client with added hook
+    And a string-flag with key "missing-flag" and a default value "uh-oh"
+    When the flag was evaluated with details
+    Then the "before" hook should have been executed
+    And the "error" hook should have been executed
+    And the "finally" hooks should be called with evaluation details
+      | data_type | key        | value          |
+      | string    | flag_key   | missing-flag   |
+      | string    | value      | uh-oh          |
+      | string    | variant    | None           |
```

Comment:
`None` is pretty python specific - i would instead use "null" which is more present in a lot of languages, compared to None and it is IMHO more descriptive